test/integration/targets/ansible-galaxy-collection-update/tasks/main.yml

```diff
@@ -0,0 +1,97 @@
+# regular expression taken from https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string
+- set_fact:
+    version_re: '((?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)(?:-(?P<prerelease>(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+(?P<buildmetadata>[0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?)'
+
+- name: test upgrading a collection and its dependencies
+  block:
+    - name: set up an upgradable collection and dependency
+      command: ansible-galaxy collection install community.aws:1.0.0 amazon.aws:1.0.0 --no-deps --force
```

Comment:
I'm sure we can update the tests once the new dep resolver changes are in but we should be using our fake Galaxy/AH endpoint rather than actual collections. Will be a lot faster to run and we can craft specific version scenarios to test out. Not sure if you've seen it before but we build a mock list of collections in this particular test target https://github.com/ansible/ansible/tree/devel/test/integration/targets/ansible-galaxy-collection.

Reply:
Yeah, I'll do that. I've seen it and avoid it (even though it's neat) because I can't use --remote.